core/node/consensus/src/storage/vm_reader.rs

```diff
@@ -0,0 +1,294 @@
+            .unwrap()
+            .clone();
+        let tx = self.gen_l2_call_tx(self.registry_address, func.short_signature().to_vec());
+
+        let res = self.eth_call(block_args, tx).await;
+        let tokens = func.decode_output(&res).context("decode_output()")?;
+        U256::from_tokens(tokens)
+            .context("U256::from_tokens()")
+            .map(|t| t.as_usize())
+    }
+
+    async fn read_committee_validator(
+        &self,
+        block_args: BlockArgs,
+        idx: usize,
+    ) -> anyhow::Result<CommitteeValidator> {
+        let func = self
+            .registry_contract
+            .function("validatorCommittee")
+            .unwrap()
+            .clone();
+        let tx = self.gen_l2_call_tx(
+            self.registry_address,
+            func.encode_input(&zksync_types::U256::from(idx).into_tokens())
+                .unwrap(),
+        );
+
+        let res = self.eth_call(block_args, tx).await;
+        let tokens = func.decode_output(&res).context("decode_output()")?;
+        CommitteeValidator::from_tokens(tokens).context("CommitteeValidator::from_tokens()")
+    }
+
+    async fn read_committee_attester(
+        &self,
+        block_args: BlockArgs,
+        idx: usize,
+    ) -> anyhow::Result<CommitteeAttester> {
+        let func = self
+            .registry_contract
+            .function("attesterCommittee")
+            .unwrap()
+            .clone();
+
+        let tx = self.gen_l2_call_tx(
+            self.registry_address,
+            func.encode_input(&zksync_types::U256::from(idx).into_tokens())
+                .unwrap(),
+        );
+
+        let res = self.eth_call(block_args, tx).await;
+        let tokens = func.decode_output(&res).context("decode_output()")?;
+        CommitteeAttester::from_tokens(tokens).context("CommitteeAttester::from_tokens()")
+    }
+
+    async fn eth_call(&self, block_args: BlockArgs, tx: L2Tx) -> Vec<u8> {
+        let call_overrides = CallOverrides {
+            enforced_base_fee: None,
+        };
+        self.tx_sender
+            .eth_call(block_args, call_overrides, tx)
+            .await
+            .unwrap()
+    }
+
+    fn gen_l2_call_tx(&self, contract_address: Address, calldata: Vec<u8>) -> L2Tx {
+        L2Tx::new(
+            contract_address,
+            calldata,
+            Nonce(0),
+            Fee {
+                gas_limit: U256::from(2000000000u32),
+                max_fee_per_gas: U256::zero(),
+                max_priority_fee_per_gas: U256::zero(),
+                gas_per_pubdata_limit: U256::from(DEFAULT_L2_TX_GAS_PER_PUBDATA_BYTE),
+            },
+            Address::zero(),
+            U256::zero(),
+            vec![],
+            Default::default(),
+        )
+    }
+}
+
+#[derive(Debug, Default)]
+#[allow(dead_code)]
+pub struct CommitteeValidator {
+    pub node_owner: Address,
+    pub weight: usize,
+    pub pub_key: Vec<u8>,
+    pub pop: Vec<u8>,
+}
+
+impl Detokenize for CommitteeValidator {
```

Comment:
There is an attribute in `ethers` that should work here even without generating the bindings. Here's an [example](https://github.com/aakoshh/ipc/blob/f13c0b6ecdbe16a18e2f3bd7eb6e898cda50552a/fendermint/vm/actor_interface/src/ipc.rs#L343-L350).